ledger/from-archive/src/main.rs

```diff
@@ -41,54 +51,242 @@ fn main() {
         )
         .expect("Could not append origin block to ledger");
 
-    // Sync all blocks
-    let mut block_index = 1;
+    let mut start = 1;
+    let mut reached_end = false;
     loop {
+        let end = start + config.block_chunk_size;
+        if reached_end {
+            log::info!(
+                logger,
+                "Done fetching transactions. Local ledger at {} blocks",
+                local_ledger.num_blocks().expect("Could not get num blocks"),
+            );
+            return;
+        }
         if let Some(block_limit) = config.num_blocks {
-            if block_index >= block_limit {
-                log::debug!(
+            if start >= block_limit {
+                log::info!(
                     logger,
                     "Done fetching transactions for {} blocks",
-                    block_index,
+                    block_limit,
                 );
                 return;
             }
         }
-        // Construct URL for the block we are trying to fetch.
-        let filename = block_num_to_s3block_path(block_index)
-            .into_os_string()
-            .into_string()
-            .unwrap();
-        let url = transactions_fetcher.source_urls[0].join(&filename).unwrap();
-
-        // Try and get the block.
-        log::debug!(
-            logger,
-            "Attempting to fetch block {} from {}",
-            block_index,
-            url
+        log::info!(logger, "Getting block contents for {}-{}", start, end);
+        let block_map = get_block_contents(
+            transactions_fetcher.clone(),
+            start.clone(),
+            end,
+            std::time::Duration::from_secs(1),
+            &logger,
         );
-        match transactions_fetcher.block_from_url(&url) {
```

Comment:
Replacing this call with `transactions_fetcher.get_block_data_by_index(block_index, None)` should be sufficient for benefiting from the merged blocks and getting the speed increase you are looking for.